storage_service/locations/api/resources.py

```diff
@@ -319,6 +319,11 @@ def save_m2m(self, bundle):
                     pipeline=item.obj,
                     location=bundle.obj)
 
+    def obj_create(self, bundle, **kwargs):
+        if 'default' in bundle.data:
+            kwargs['default'] = bundle.data['default']
+        return super(LocationResource, self).obj_create(bundle, **kwargs)
+
```

Comment:
It was a bit subtle to me that `obj_create` calls `for key, value in kwargs.items(): setattr(bundle.obj, key, value)` in tastypie/resources.py, which calls the `default` attr's setter on the location, which ultimately results in `set_default_location_post_save` creating the `Setting` instance to store the default nature of the location.

Reply:
Yeah, not great. The "default" attribute is fairly simple at the model layer but I could probably do better in `LocationResource` (tastypie). I remember trying to add `default` as a new filed in `LocationResource` hoping that'd make things much cleaner but I had some issues.

Reply:
I've added a comment to the method hoping that will make things easier to understand for the reader.